src/services/ActiveSolutionDiagramService.ts

```diff
@@ -0,0 +1,32 @@
+import {IDiagram} from '@process-engine/solutionexplorer.contracts';
+import {ISolutionEntry, ISolutionService} from '../contracts';
+
+export class SolutionService implements ISolutionService {
+  private _allSolutionEntries: Array<ISolutionEntry> = [];
+  private _activeSolution: ISolutionEntry;
```

Comment:
```suggestion
  private _activeSolutionReference: ISolutionEntry;
```

As this should be no copy of the active solution, but instead the reference to one in the list!